plugin/src/org/jetbrains/cabal/psi/VersionConstraint.kt

```diff
@@ -61,15 +61,14 @@ public class VersionConstraint(node: ASTNode) : ASTWrapperPsiElement(node), Chec
 
     public fun isSimple(): Boolean = !(getVersionValue() endsWith '*')
 
-    public fun getComparator(): String? = COMPARATORS firstOrNull { it.equals((this : PsiElement).getFirstChild()!!.getText()!!) }
+    public fun getComparator(): String? = COMPARATORS firstOrNull { it.equals(this.getFirstChild()!!.getText()!!) }
 
-    public fun getVersion(): String = (this : PsiElement).getLastChild()!!.getText()!!
+    public fun getVersion(): String = this.getLastChild()!!.getText()!!
 
     public fun getVersionValue(): String = (getVersion().replace("(\\-[0-9a-zA-Z]+)+\\.".toRegex(), "\\.")).replace("(\\-[0-9a-zA-Z]+)+$".toRegex(), "")
 
     public override fun check(): List<ErrorMessage> {
-        val comparator = getComparator()
-        if (comparator == null) return listOf()
+        val comparator = getComparator() ?: return listOf()
```

Comment:
This seems to be an equivalent expression. Are you sure the previous one caused a compilation error?